source/extensions/omni.isaac.lab/omni/isaac/lab/assets/deformable_object/deformable_object.py

```diff
@@ -0,0 +1,413 @@
+
+import omni.isaac.lab.sim as sim_utils
+import omni.isaac.lab.utils.math as math_utils
+from omni.isaac.lab.markers import VisualizationMarkers
+
+from ..asset_base import AssetBase
+from .deformable_object_data import DeformableObjectData
+
+if TYPE_CHECKING:
+    from .deformable_object_cfg import DeformableObjectCfg
+
+
+class DeformableObject(AssetBase):
+    """A deformable object asset class.
+
+    Deformable objects are assets that can be deformed in the simulation. They are typically used for
+    soft bodies, such as stuffed animals and food items.
+
+    Unlike rigid object assets, deformable objects have a more complex structure and require additional
+    handling for simulation. The simulation of deformable objects follows a finite element approach, where
+    the object is discretized into a mesh of nodes and elements. The nodes are connected by elements, which
+    define the material properties of the object. The nodes can be moved and deformed, and the elements
+    respond to these changes.
+
+    The state of a deformable object comprises of its nodal positions and velocities, and not the object's root
+    position and orientation. The nodal positions and velocities are in the simulation frame.
+
+    Soft bodies can be `partially kinematic`_, where some nodes are driven by kinematic targets, and the rest are
+    simulated. The kinematic targets are the desired positions of the nodes, and the simulation drives the nodes
+    towards these targets. This is useful for partial control of the object, such as moving a stuffed animal's
+    head while the rest of the body is simulated.
+
+    .. attention::
+        This class is experimental and subject to change due to changes on the underlying PhysX API on which
+        it depends. We will try to maintain backward compatibility as much as possible but some changes may be
+        necessary.
+
+    .. _partially kinematic: https://nvidia-omniverse.github.io/PhysX/physx/5.4.1/docs/SoftBodies.html#kinematic-soft-bodies
+    """
+
+    cfg: DeformableObjectCfg
+    """Configuration instance for the deformable object."""
+
+    def __init__(self, cfg: DeformableObjectCfg):
+        """Initialize the deformable object.
+
+        Args:
+            cfg: A configuration instance.
+        """
+        super().__init__(cfg)
+
+    """
+    Properties
+    """
+
+    @property
+    def data(self) -> DeformableObjectData:
+        return self._data
+
+    @property
+    def num_instances(self) -> int:
+        return self.root_physx_view.count
+
+    @property
+    def num_bodies(self) -> int:
+        """Number of bodies in the asset.
+
+        This is always 1 since each object is a single deformable body.
+        """
+        return 1
+
+    @property
+    def root_physx_view(self) -> physx.SoftBodyView:
+        """Deformable body view for the asset (PhysX).
+
+        Note:
+            Use this view with caution. It requires handling of tensors in a specific way.
```

Comment:
NIT: This is very vague, but I think it's copy / pasted from other assets?

Reply:
Yup. But I think we can remove this note everywhere. It was only back in the days where we had to handle indexing of tensors ourselves. I think users can now do it as well since it isn't so complicated.